instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt

```diff
@@ -905,6 +908,58 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
             .first()
     }
 
+    @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+    @Test
+    fun routeRefreshMultipleAttemptsWhenAlternativesChange() = sdkTest {
+        val refreshStates = mutableListOf<RouteRefreshStateResult>()
+        val statesObserver = RouteRefreshStatesObserver {
+            refreshStates.add(it)
+        }
+        mapboxNavigation.routeRefreshController.registerRouteRefreshStateObserver(statesObserver)
+
+        val routeOptions = generateRouteOptions(twoCoordinates)
+
+        mockWebServerRule.requestHandlers.remove(failByRequestRouteRefreshResponse)
+        mockWebServerRule.requestHandlers.add(
+            NthAttemptHandler(
+                MockDirectionsRefreshHandler(
+                    "route_response_route_refresh",
+                    readRawFileText(activity, R.raw.route_response_route_refresh_annotations),
+                    routeIndex = 0
+                ),
+                1
+            )
+        )
+
+        val routes = mapboxNavigation.requestRoutes(routeOptions)
+            .getSuccessfulResultOrThrowException()
+            .routes
+        val initialRoutes = routes.take(1)
+
+        mapboxNavigation.startTripSession()
+        stayOnInitialPosition()
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(initialRoutes)
+
+        delay(5000) // 3s refresh interval + 2s to be in-between
+
+        // set alternatives
+        mapboxNavigation.setNavigationRoutesAndWaitForAlternativesUpdate(routes)
+
+        val refreshedRoutes = withTimeout(2000) { // refresh interval is 3s, 1s accuracy
+            mapboxNavigation.routesUpdates().first { it.reason == ROUTES_UPDATE_REASON_REFRESH }
+        }
+        assertEquals(routes.map { it.id }, refreshedRoutes.navigationRoutes.map { it.id })
+        assertEquals(
+            listOf(
+                RouteRefreshExtra.REFRESH_STATE_STARTED,
+                RouteRefreshExtra.REFRESH_STATE_CANCELED,
+                RouteRefreshExtra.REFRESH_STATE_STARTED,
+                RouteRefreshExtra.REFRESH_STATE_FINISHED_SUCCESS,
+            ),
```

Comment:
given the first delay of 5 seconds and the second of 2 seconds and refresh interval 3 seconds, so that 2 route refresh happen, shouldn't states contain two `REFRESH_STATE_FINISHED_SUCCESS`?

Reply:
Oh, I didn't notice `NthAttemptHandler`